pkg/action/list_test.go

```diff
@@ -286,6 +288,32 @@ func makeMeSomeReleases(store *storage.Storage, t *testing.T) {
 	assert.Len(t, all, 3, "sanity test: three items added")
 }
 
+func makeMeSomeReleasesWithDifferentNamespaces(store *storage.Storage, t *testing.T) {
+	t.Helper()
+	one := releaseStub()
+	one.Name = "one"
+	one.Namespace = "default1"
+	one.Version = 1
+	two := releaseStub()
+	two.Name = "two"
+	two.Namespace = "default2"
+	two.Version = 2
+	three := releaseStub()
+	three.Name = "three"
+	three.Namespace = "default3"
+	three.Version = 3
+
+	for _, rel := range []*release.Release{one, two, three} {
+		if err := store.Create(rel); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	_, err := store.ListDeployed()
+	assert.NoError(t, err)
+	//assert.Len(t, all, 3, "sanity test: three items added")
+}
+
```

Comment:
Regarding testing, I'm not that familiar with go to know what's possible, but some ideas are:
Test if AllNamespaces utilizes a cloned cfg.
Test if original cfg is used/accessed after the cloning has happened (make sure devs don't accidently use l.cfg instead of suggested configuration := l.cfg).
Test if original cfg is still the same after Run().
Test if all the driver Name() strings map to Configuration.Init(helmDriver string)

Reply:
I have added a test to check if we get all the releases in all namespaces and testing if the config is cloned.